File: packages/bot-engine/src/strategies/Scouter.js
import { RSI } from 'technicalindicators';

export const ScouterStrategy = {
  name: 'AI_SCOUTER',
  description: '🏹 Scouter Scalping Strategy',
  
  /**
   * Compute signal based on SMA and RSI.
   * @param {number[]} closes 
   * @param {object} params 
   */
  compute: (closes, params = {}) => {
    const sma7 = closes.slice(-7).reduce((a, b) => a + b, 0) / 7;
    const sma14 = closes.slice(-14).reduce((a, b) => a + b, 0) / 14;
    
    const rsiValues = RSI.calculate({ period: 14, values: closes });
    const rsi = rsiValues.length > 0 ? rsiValues.at(-1) : 50;
    
    if (sma7 > sma14 && rsi < 55) return 'LONG';
    if (sma7 < sma14 && rsi > 45) return 'SHORT';
    
    return 'NONE';
  },

  /**
   * Human readable reason for the entry.
   */
  describe: (signal) => {
    return `🏹 สัญญาณ Scalping จาก AI_SCOUTER (${signal})`;
  },

  /**
   * Diagnostic thought for AI_SCOUTER
   */
  getDiagnostic: (closes) => {
    if (closes.length < 14) return "กำลังรวบรวมข้อมูลราคา (EMA/RSI)...";
    const sma7 = closes.slice(-7).reduce((a, b) => a + b, 0) / 7;
    const sma14 = closes.slice(-14).reduce((a, b) => a + b, 0) / 14;
    const rsiValues = RSI.calculate({ period: 14, values: closes });
    const rsi = rsiValues.length > 0 ? rsiValues.at(-1) : 50;

    const rsiText = rsi.toFixed(1);
    const trend = sma7 > sma14 ? "Bullish (SMA7 > SMA14)" : "Bearish (SMA7 < SMA14)";
    
    if (sma7 > sma14 && rsi >= 55) return `ตลาด: ขาขึ้น | RSI: ${rsiText} (> 55) - รอ RSI ย่อตัวเพื่อเข้า LONG`;
    if (sma7 < sma14 && rsi <= 45) return `ตลาด: ขาลง | RSI: ${rsiText} (< 45) - รอ RSI ดีดตัวเพื่อเข้า SHORT`;
    
    return `เทรนด์: ${trend} | RSI: ${rsiText} - กำลังสแกนหาจุด Scalping`;
  }
};

File: packages/strategy-ai/strategies/scouter.py
import pandas as pd
import ta
from base_strategy import BaseStrategy


class ScouterStrategy(BaseStrategy):
    """AI Scouter — SMA7/14 + RSI scalping — powered by ta"""

    def populate_indicators(self, df: pd.DataFrame, params: dict) -> pd.DataFrame:
        df["sma7"] = ta.trend.sma_indicator(df["close"], window=7)
        df["sma14"] = ta.trend.sma_indicator(df["close"], window=14)
        df["rsi"] = ta.momentum.rsi(df["close"], window=14)
        return df

    def populate_signals(self, df: pd.DataFrame, params: dict) -> pd.DataFrame:
        df["signal"] = "NONE"
        df.loc[(df["sma7"] > df["sma14"]) & (df["rsi"] < 55), "signal"] = "LONG"
        df.loc[(df["sma7"] < df["sma14"]) & (df["rsi"] > 45), "signal"] = "SHORT"
        return df

    def get_metadata(self) -> dict:
        return {"name": "AI_SCOUTER", "description": "SMA7/14 + RSI Scouter", "version": "3.0.0"}
